src/Spanner/Database.php

```diff
@@ -0,0 +1,1313 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner;
+
+use Google\Cloud\Core\Exception\AbortedException;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\Iam\Iam;
+use Google\Cloud\Core\LongRunning\LongRunningOperation;
+use Google\Cloud\Core\LongRunning\LROTrait;
+use Google\Cloud\Core\LongRunning\LongRunningConnectionInterface;
+use Google\Cloud\Core\Retry;
+use Google\Cloud\Spanner\Connection\ConnectionInterface;
+use Google\Cloud\Spanner\Connection\IamDatabase;
+use Google\Cloud\Spanner\Session\Session;
+use Google\Cloud\Spanner\Session\SessionPoolInterface;
+use Google\Cloud\Spanner\V1\SpannerClient as GrpcSpannerClient;
+
+/**
+ * Represents a Google Cloud Spanner Database.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Spanner\SpannerClient;
+ *
+ * $spanner = new SpannerClient();
+ *
+ * $database = $spanner->connect('my-instance', 'my-database');
+ * ```
+ *
+ * ```
+ * // Databases can also be connected to via an Instance.
+ * use Google\Cloud\Spanner\SpannerClient;
+ *
+ * $spanner = new SpannerClient();
+ *
+ * $instance = $spanner->instance('my-instance');
+ * $database = $instance->database('my-database');
+ * ```
+ */
+class Database
+{
+    use LROTrait;
+    use TransactionConfigurationTrait;
+
+    const MAX_RETRIES = 3;
+
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var Instance
+     */
+    private $instance;
+
+    /**
+     * @var LongRunningConnectionInterface
+     */
+    private $lroConnection;
+
+    /**
+     * @var Operation
+     */
+    private $operation;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var Iam
+     */
+    private $iam;
+
+    /**
+     * @var Session|null
+     */
+    private $session;
+
+    /**
+     * @var SessionPoolInterface|null
+     */
+    private $sessionPool;
+
+    /**
+     * Create an object representing a Database.
+     *
+     * @param ConnectionInterface $connection The connection to the
+     *        Google Cloud Spanner Admin API.
+     * @param Instance $instance The instance in which the database exists.
+     * @param LongRunningConnectionInterface $lroConnection An implementation
+     *        mapping to methods which handle LRO resolution in the service.
+     * @param string $projectId The project ID.
+     * @param string $name The database name.
```

Comment:
Either a name or an ID is accepted. Internally, an ID will be provided. The convention throughout most of the library is to use the term "name". Since either a name or an ID is acceptable here, is it ok that we continue to use that term, or would you still prefer I change the name of those argument to `id`?